controller/engine_controller.go

```diff
@@ -1803,6 +1810,13 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 			}
 			logrus.Debug("Started snapshot purge after rebuilding")
 		}
+
+		// check and reset replica rebuild failed condition
+		if err := updateReplicaRebuildFailedCondition(log, ec.ds, replica, ""); err != nil {
+			log.WithError(err).Errorf("Failed to update rebuild status information on replica %v", replica)
+			return
+		}
```

Comment:
This should be moved to the beginning when starting rebuilding to avoid any leftover condition still when the replica is rebuilding.

Reply:
Moved to the beginning when starting rebuilding. Thanks